use-shopping-cart/core/Entry.js

```diff
@@ -0,0 +1,199 @@
+import { formatCurrencyString } from './store'
+import { v4 as uuidv4 } from 'uuid'
+
+function Entry({
+  product,
+  quantity,
+  currency,
+  language,
+  price_metadata,
+  product_metadata
+}) {
+  quantity = parseInt(quantity)
+  const id =
+    product.id || product.price_id || product.sku_id || product.sku || uuidv4()
+
+  const productCopy = { ...product }
+
+  if (!productCopy.price_data && price_metadata) {
+    productCopy.price_data = {
+      ...price_metadata
+    }
+  } else if (productCopy.price_data && price_metadata) {
+    productCopy.price_data = {
+      ...productCopy.price_data,
+      ...price_metadata
+    }
+  }
+
+  if (!productCopy.product_data && product_metadata) {
+    productCopy.product_data = {
+      ...product_metadata
+    }
+  } else if (productCopy.product_data && product_metadata) {
+    productCopy.product_data = {
+      ...productCopy.product_data,
+      ...product_metadata
+    }
+  }
+
+  return {
+    ...productCopy,
+    id,
+    quantity,
+    value: product.price * quantity,
+    formattedValue: formatCurrencyString({
+      value: product.price * quantity,
+      currency,
+      language
+    })
```

Comment:
The way you're doing this every single time seems entirely unnecessary since getters are serializable and if you do it this way, you're running `formatCurrencyString` every single time something changes instead of just when `formattedValue` is accessed. A getter would just make more sense in this situation in my opinion.

Reply:
So I tried doing the getter like before, but I struggled with it because Immer didn't allow for it. Can't remember the specific error, but I'm going to try to use getters here and report back. I definitely agree with the getter, I just couldn't get it to cooperate

Reply:
![Screen Shot 2021-03-11 at 10 31 00 PM](https://user-images.githubusercontent.com/1852675/110888082-75b09700-82b9-11eb-8e5f-07d0d61aac37.png)


@ChrisBrownie55 this is why I removed the getter, I have no idea how to get around this error in particular. I saw one solution was to use a JS method to add a getter with an object assignment method but Immer doesn't like that either. Any advice would be appreciated 🙏

Reply:
I wasn't able to reproduce this error in a sandbox. Can you show me what you put for the getter?

Reply:
Here is my sandbox I tried it in:

https://codesandbox.io/s/redux-getters-0y3xi

Reply:
Oh, I did NOT try doing it like this. I'll try adding these in